internal/repositories/dcn.go

```diff
@@ -59,3 +63,141 @@ func (r *Repository) GetDCNByName(ctx context.Context, name string) (*gmodel.Dcn
 
 	return DCNToAPI(dcn), nil
 }
+
+func (r *Repository) GetDCNs(ctx context.Context, first *int, after *string, last *int, before *string, filterBy *gmodel.DCNFilter) (*gmodel.DCNConnection, error) {
+	var limit int
+
+	if first != nil {
+		if last != nil {
+			return nil, gqlerror.Errorf("Pass `first` or `last`, but not both.")
+		}
+		if *first < 0 {
+			return nil, gqlerror.Errorf("The value for `first` cannot be negative.")
+		}
+		if *first > maxPageSize {
+			return nil, gqlerror.Errorf("The value %d for `first` exceeds the limit %d.", *last, maxPageSize)
+		}
+		limit = *first
+	} else {
+		if last == nil {
+			return nil, gqlerror.Errorf("Provide `first` or `last`.")
+		}
+		if *last < 0 {
+			return nil, gqlerror.Errorf("The value for `last` cannot be negative.")
+		}
+		if *last > maxPageSize {
+			return nil, gqlerror.Errorf("The value %d for `last` exceeds the limit %d.", *last, maxPageSize)
+		}
+		limit = *last
+	}
+
+	queryMods := []qm.QueryMod{}
+
+	if filterBy != nil && filterBy.Owner != nil {
+		queryMods = append(queryMods, models.DCNWhere.OwnerAddress.EQ(filterBy.Owner.Bytes()))
+	}
+
+	dcnCount, err := models.DCNS(queryMods...).Count(ctx, r.pdb.DBS().Reader)
+	if err != nil {
+		return nil, err
+	}
+
+	orderBy := " DESC"
+	if last != nil {
+		orderBy = " ASC"
+	}
+
+	queryMods = append(queryMods, []qm.QueryMod{
+		qm.Limit(limit + 1),
+		qm.OrderBy(models.DCNColumns.MintedAt + orderBy),
+	}...)
```

Comment:
https://go.dev/play/p/_Vs3xR_u8l0

Reply:
Here's the general area

https://github.com/DIMO-Network/identity-api/blob/c4e1456c6969e4f2a4cc89ec90c8465702d7ee69/internal/repositories/vehicle_privileges.go#L119